ui/app/components/mount-backend-form.ts

```diff
@@ -191,6 +191,13 @@ export default class MountBackendForm extends Component<Args> {
     this.checkPathChange(value);
   }
 
+  @action
+  resetForm(value: string) {
+    // clears the identityTokenKey and path, not the entire form.
```

Comment:
Would argue the flow here could be improved upon, but we do currently handle the full "teardown" in [willDestroy](https://github.com/hashicorp/vault/pull/28822/files#diff-41db6b6e2124359c99baa219be682e398203fab9bc7cce398b764f221079f2b3R51). I am simply keeping the previous behavior and adding an extra check for removing the identity_token_key if a user presses the back button.

I suspect because of the `belongs-to` relationship on the `secret-engine` model, unloading the "config" attr on this model throws an error. Ideally, this is what we'd like to do, but for now, I'm just resetting the identity_token_key and path.

Reply:
So this is new functionality since right now we don't clear the path value when clicking the back button. Is this intended?

If we want to keep this method I think we should make it more flexible and not pass a value here. A couple options are just calling `this.resetForm` and then inside this function we could do `this.handleIdentityTokenKeyChange = ''` Then we could expand it to include additional reset values in the future..

However, I'm not entirely sure we need this functionality...looking at ember data, the param is removed when clicking "back" as it is so it feels superfluous

Reply:
We do clear the path when they press the back button though it's not super clear here. It was on the hbs file. See [here](https://github.com/hashicorp/vault/blob/main/ui/app/components/mount-backend-form.hbs#L67). 

And I had to add this function in because it was failing on a test and then I confirmed locally. If you fill in the IdentityTokenKey and click "Cancel" then choose another Secret engine with this option (say aws and then azure) you'll see it filled in.

Reply:
That line you pointed to clears the `mountType` not the mount `path`. 

Maybe that's a UX question then - if I was a user and had filled out a form, then meant to select a different WIF engine, I'd want that data to be saved and not cleared when switching to a different engine type. I think this is an edge case, so probably not worth delving into too much. Just my two cents!

Reply:
Ah, thank you for the clarification. Let me remove this for now, return to previous functionality and then ask design as a follow up.